common/app/renderers/DotcomRenderingService.scala

```diff
@@ -218,29 +214,27 @@ class DotcomRenderingService extends GuLogging with ResultWithPreconnectPreload
   private def baseArticleRequest(
       path: String,
       ws: WSClient,
-      page: PageWithStoryPackage,
-      blocks: Blocks,
+      pageBlocks: BlocksOn[PageWithStoryPackage],
       pageType: PageType,
       filterKeyEvents: Boolean,
       forceLive: Boolean = false,
       newsletter: Option[NewsletterData],
   )(implicit request: RequestHeader): Future[Result] = {
-    val dataModel = page match {
+    val dataModel = pageBlocks.page match {
       case liveblog: LiveBlogPage =>
         DotcomRenderingDataModel.forLiveblog(
-          liveblog,
-          blocks,
+          pageBlocks.copy(page = liveblog),
```

Comment:
Is it the case that the match confirms that `pageBlocks` should already be of type `BlocksOn[LiveBlogPage]`, but the compiler isn't able to narrow it, so this copy is a way to get around that? You can't do something like:

```scala
pageBlocks match {
  case liveblogPageBlocks: BlocksOn[LiveBlogPage] =>
    DotcomRenderingDataModel.forLiveblog(
      liveblogPageBlocks,
      ...
    )
}
```

due to type erasure?